modules/ducktests/tests/ignitetest/services/utils/control_utility.py

```diff
@@ -0,0 +1,167 @@
+        """
+        :param nodes: Nodes that should be removed to baseline.
+        """
+        result = self.__run("--baseline remove %s --yes" %
+                            ",".join([node.account.externally_routable_ip for node in nodes]))
+
+        return self.__parse_cluster_state(result)
+
+    def disable_baseline_auto_adjust(self):
+        """
+        Disable baseline auto adjust.
+        """
+        return self.__run("--baseline auto_adjust disable --yes")
+
+    def enable_baseline_auto_adjust(self, timeout=None):
+        """
+        Enable baseline auto adjust.
+        :param timeout: Auto adjust timeout in millis.
+        """
+        timeout_str = "timeout %d" % timeout if timeout else ""
+        return self.__run("--baseline auto_adjust enable %s --yes" % timeout_str)
+
+    def activate(self):
+        """
+        Activate cluster.
+        """
+        return self.__run("--activate --yes")
+
+    def deactivate(self):
+        """
+        Deactivate cluster.
+        """
+        return self.__run("--deactivate --yes")
+
+    @staticmethod
+    def __parse_cluster_state(output):
+        state_pattern = re.compile("Cluster state: ([^\\s]+)")
+        topology_pattern = re.compile("Current topology version: (\\d+)")
+        baseline_pattern = re.compile("Consistent(Id|ID)=([^\\s]+),\\sS(tate|TATE)=([^\\s]+),?(\\sOrder=(\\d+))?")
+
+        match = state_pattern.search(output)
+        state = match.group(1) if match else None
+
+        match = topology_pattern.search(output)
+        topology = int(match.group(1)) if match else None
+
+        baseline = [BaselineNode(consistent_id=m[1], state=m[3], order=int(m[5]) if m[5] else None)
+                    for m in baseline_pattern.findall(output)]
+
+        return ClusterState(state=state, topology_version=topology, baseline=baseline)
+
+    def __run(self, cmd):
+        node = random.choice(self.__alives())
+
+        self.logger.debug("Run command %s on node %s", cmd, node.name)
+
+        raw_output = node.account.ssh_capture(self.__form_cmd(node, cmd), allow_fail=True)
+        code, output = self.__parse_output(raw_output)
+
+        self.logger.debug("Output of command %s on node %s, exited with code %d, is %s", cmd, node.name, code, output)
+
+        if code != 0:
+            raise ControlUtilityError(node.account, cmd, code, output)
+
+        return output
+
+    def __form_cmd(self, node, cmd):
+        return self._cluster.path.script("%s --host %s %s" % (self.BASE_COMMAND, node.account.externally_routable_ip,
+                                                              cmd))
+
+    @staticmethod
+    def __parse_output(raw_output):
+        exit_code = raw_output.channel_file.channel.recv_exit_status()
+        output = "".join(raw_output)
+
+        pattern = re.compile("Command \\[[^\\s]*\\] finished with code: (\\d+)")
+        match = pattern.search(output)
+
+        if match:
+            return int(match.group(1)), output
```

Comment:
Is it ok, that it's possible to receive `exit_code !=0` but in the same time text output will have code equals to 0?